builtin/logical/nomad/backend.go

```diff
@@ -0,0 +1,74 @@
+package nomad
+
+import (
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func Factory(conf *logical.BackendConfig) (logical.Backend, error) {
+	b := Backend()
+	if err := b.Setup(conf); err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func Backend() *backend {
+	var b backend
+	b.Backend = &framework.Backend{
+		Paths: []*framework.Path{
+			pathConfigAccess(&b),
+			pathConfigLease(&b),
+			pathListRoles(&b),
+			pathRoles(&b),
+			pathCredsCreate(&b),
+		},
+
+		Secrets: []*framework.Secret{
+			secretToken(&b),
+		},
+		BackendType: logical.TypeLogical,
+	}
+
+	return &b
+}
+
+type backend struct {
+	*framework.Backend
+}
+
+func (b *backend) client(s logical.Storage) (*api.Client, error) {
+	conf, err := b.readConfigAccess(s)
+	if err != nil {
+		return nil, err
+	}
+
+	nomadConf := api.DefaultConfig()
+	nomadConf.Address = conf.Address
```

Comment:
The `address` field is not enforced in the configuration. This can overwrite the address in the default configuration by an empty string. If the address is supposed to be overwritten here every time, can we make it mandatory on the config/access endpoint?

Reply:
Fixed in a28088443, making it mandatory to provide an address in config/access

Reply:
If env vars are being used to provide address and token, the call above would end up returning a nil client, and even if it didn't, this would then override the env vars with blanks.

Rather than making it mandatory to provide access config, you should switch this around:
```go
conf, err := b.readConfigAccess(s)
if err != nil {
	return nil, err
}

nomadConf := api.DefaultConfig()

if conf != nil {
	if conf.Address != "" {
		nomadConf.Address = conf.Address
	}
	if conf.Token != "" {
		nomadConf.SecretID = conf.Token
	}
}
```
Optionally after this you may want to check to make sure the Address and SecretID in nomadConf are populated and return an err if so; I'm not sure if that's appropriate or not with the Nomad libs.

Also, if the official name for Nomad access tokens are Secret IDs, you may want to rename the conf object to match instead of using Token.